test/modules/TestResultCoordinator/Reports/EdgeHubRestartTest/EdgeHubRestartDirectMethodReportGenerator.cs

```diff
@@ -0,0 +1,303 @@
+                    (senderResultCount, hasSenderResult, senderSeqNum) = await this.IncrementSequenceNumberAsync(
+                        hasSenderResult,
+                        this.MoveNextSenderResultAsync,
+                        receiverSeqNum,
+                        senderResultCount);
+                }
+
+                if (receiverSeqNum < senderSeqNum)
+                {
+                    // Increment receiver result to have the same seq as the sender
+                    (receiverResultCount, hasReceiverResult, receiverSeqNum) = await this.IncrementSequenceNumberAsync(
+                        hasReceiverResult,
+                        this.MoveNextReceiverResultAsync,
+                        senderSeqNum,
+                        receiverResultCount);
+                }
+
+                if (hasSenderResult ^ hasReceiverResult)
+                {
+                    break;
+                }
+
+                // Note: In the current verification, if the receiver obtained more result dm than
+                //   sender, the test is consider a fail. We are disregarding duplicated/unique seqeunce number
+                //   of receiver's result.
+
+                // Check if the current dm is passing
+                bool isCurrentDirectMethodPassing = true;
+                EdgeHubRestartDirectMethodResult senderResult = JsonConvert.DeserializeObject<EdgeHubRestartDirectMethodResult>(this.SenderTestResults.Current.Result);
+
+                // Verified the sequence numbers are the same
+                isCurrentDirectMethodPassing &= senderSeqNum == receiverSeqNum;
+
+                // Verified the sequence numbers are incremental
+                isCurrentDirectMethodPassing &= senderSeqNum > previousSeqNum;
+
+                // Make sure the status code is passing
+                isCurrentDirectMethodPassing &= senderResult.DirectMethodCompletedStatusCode == HttpStatusCode.OK;
+
+                // If the current DM result is passed, increment the count for a good dm
+                passedDirectMethodCount += isCurrentDirectMethodPassing ? 1UL : 0UL;
+                previousSeqNum = senderSeqNum;
+
+                (senderResultCount, hasSenderResult, senderSeqNum) = await this.MoveNextSenderResultAsync(senderResultCount);
+                (receiverResultCount, hasReceiverResult, receiverSeqNum) = await this.MoveNextReceiverResultAsync(receiverResultCount);
+            }
+
+            (senderResultCount, _, _) = await this.IncrementSequenceNumberAsync(
+                hasSenderResult,
+                this.MoveNextSenderResultAsync,
+                long.MaxValue,
+                senderResultCount);
+
+            (receiverResultCount, _, _) = await this.IncrementSequenceNumberAsync(
+                hasReceiverResult,
+                this.MoveNextReceiverResultAsync,
+                long.MaxValue,
+                receiverResultCount);
+
+            return this.CalculateStatistic(
```

Comment:
Rename this method name; which it turns a test result report, not statistic.  Just write statistic to log file, and put into the report.